### PR TITLE
feature/devise-login

### PR DESCRIPTION
**Pull request by Rodrigo-Miranda-0**
#### Board:
* [Ticket #WcuUqEx2](https://trello.com/c/WcuUqEx2/14-3-as-a-user-i-should-be-able-to-login-so-i-can-use-my-target-mvd-account)
---
#### Description:

* Added corresponding devise gem login tests.
---
#### Notes:
*
---
#### Tasks:
- [x] Add login functionality/tests
---
#### Risk:
*
---
#### Preview:

### CODE REVIEW

File: spec/requests/api/v1/users/login_spec.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+require "rspec/json_expectations"
+
+describe 'Login User', type: :request do
+  subject { post '/api/v1/auth/sign_in', params:, as: :json }
+
+  before(:each) do
+    @current_user = FactoryBot.create(:user)
+  end
```
Comment (juanmferreira93) [inline spec/requests/api/v1/users/login_spec.rb]:
```suggestion
  let!(:user) { create(:user) } 
```
then you should replace the `@current_user` for `user` in each place

File: spec/requests/api/v1/users/login_spec.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+require "rspec/json_expectations"
+
+describe 'Login User', type: :request do
+  subject { post '/api/v1/auth/sign_in', params:, as: :json }
```
Comment (juanmferreira93) [inline spec/requests/api/v1/users/login_spec.rb]:
```suggestion
  subject { post api_v1_users_sign_in_path, params:, as: :json }
```
Check if your route match with that, but the idea is to use rails paths

File: spec/requests/api/v1/users/login_spec.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      end
+
+      it 'should return the error message (Invalid credentials)' do
+        subject
+        expect(response.body).to include_json(
+          errors: [
+            "Invalid login credentials. Please try again."
+          ]
+        )
+      end
```
Comment (juanmferreira93) [inline spec/requests/api/v1/users/login_spec.rb:53]:
```suggestion
  context 'when fails' do
    context 'with invalid credentials' do
      let(:params) do
        {
          email: user.email,
          password: "12345678"
        }
      end

      it 'return an unprocessable entity status' do
        subject
        expect(response).to have_http_status(:unauthorized)
      end

      it 'return the error message (Invalid credentials)' do
        subject
        expect(response.body).to include_json(
          errors: [
            "Invalid login credentials. Please try again."
          ]
        )
      end
```
Just some minor comments that you can fix in another PR :)

**Review by juanmferreira93 — APPROVED** [review]

**Review by enzofab91 — APPROVED** [review]
🌝

File: spec/requests/api/v1/users/login_spec.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
+require 'rails_helper'
+require "rspec/json_expectations"
```
Comment (jpascual1994) [inline spec/requests/api/v1/users/login_spec.rb:2]: Maybe we should move this to the `spec_helper` so it's available for every test and we don't have to include it every time

**Review by jpascual1994 — APPROVED** [review]
LGTM

Just a minor comment/suggestion

**Pull request merged**